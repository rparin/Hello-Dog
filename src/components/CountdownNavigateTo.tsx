"use client";
import { useRouter } from "next/navigation";
import { useCountdown } from "@/hooks/useCountdown";

export default function CountdownNavigateTo(props: {
  href: string;
  countdown: number;
  className?: string;
}) {
  const router = useRouter();
  const ct = useCountdown(props.countdown, () => {
    router.push(props.href);
  });
  return <span className={props.className}>{ct / 1000}</span>;
}
