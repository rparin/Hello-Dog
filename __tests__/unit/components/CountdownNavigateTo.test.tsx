import { act, render } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import CountdownNavigateTo from "@/components/CountdownNavigateTo";

expect.extend(toHaveNoViolations);

test("should have no accessibility violations", async () => {
  const { container } = render(
    <CountdownNavigateTo href={"/"} countdown={0} />
  );
  const results = await act(async () => axe(container));
  expect(results).toHaveNoViolations();
});

test("it renders component unchanged", () => {
  const { container } = render(
    <CountdownNavigateTo href={"/"} countdown={0} />
  );
  expect(container).toMatchSnapshot();
});
