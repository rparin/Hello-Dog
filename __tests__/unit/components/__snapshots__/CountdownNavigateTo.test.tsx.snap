// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders component unchanged 1`] = `
<div>
  <span>
    0
  </span>
</div>
`;
